src/peer_manager.rs

```diff
@@ -962,14 +967,40 @@ impl PeerManager {
             .count()
     }
 
+    /// Updates the given clients total traffic amount.
+    pub fn add_client_traffic(&mut self, pub_id: &PublicId, added_bytes: u64) {
+        let self_pfx = format!("{:?}", self);
+        let _ = self.peers
+            .get_mut(pub_id)
+            .map(|peer| if let PeerState::Client {
+                            ip,
+                            traffic: old_traffic,
+                        } = *peer.state() {
+                     let new_traffic = old_traffic + added_bytes;
```

Comment:
i know the chance will be very tiny as we are using u64 for the traffic.
however, shall we still consider using `wrapping_add` ?